More Dev3!

SwiffterAnim opened the pull request:
Added new Enemy!
Electric Wall

SwiffterAnim committed:
Added new Enemy!

Electric Wall

omidizadi commented on Assets/Modules/Enemies/ElectricWallStarter.cs:
```diff
@@ -0,0 +1,59 @@
+
+    private IEnumerator DestroyElectricWall()
+    {
+        yield return new WaitForSeconds(17);
```
Same as beam frequency, let's make this 17 a global variable

SwiffterAnim replied:
done

omidizadi commented on Assets/Modules/Enemies/Sprites/Smoke-3-Sheet.png:
Question: Did you handpaint this 2d effects or did you use a specific software/tool? :D I'm curious!

SwiffterAnim replied:
I animated it in procreate on my ipad =)

omidizadi replied:
Nice! Great job :D

omidizadi commented on Assets/Modules/Enemies/EnemySpawnManager.cs:
```diff
@@ -1,38 +1,78 @@
+    private GameObject shooterPrefab;
 
-    private float timer;
+    public bool isSpawningShooter;
```
Where are you assigning the value of this boolean? I have the same argument as "IsAlive" boolean in PlayerManager. This should be only readable for other classes and they shouldn't be able to change its value

SwiffterAnim replied:
For now, this one, I'm assigining it myself in the inspector, don't have any other script touching this. I would perhaps, on game manager, or difficulty manager, to have the EnemySpawnManager stop or start instancing certain types of enemies depending on the level/difficulty later.

omidizadi replied:
oh then if you're using it in the inspector it's better to make it:

[Serializefield] private bool isSpawningShooter;

isAlive case was a bit different than this sorry for confusing you 😅

SwiffterAnim committed:
Struggle with Coroutines

Was trying to switch to coroutines in the Enemy Spawn Manager, and struggled a bit with the concept. I think I made it work now, but it took me a while hehe
